composables/useBrowserExtension.ts

```diff
@@ -0,0 +1,136 @@
+import { PROVIDERS, STORAGE_KEY } from '@/types/enums'
+import { profileRoute } from '@/shared/routes'
+import { INJECTED_PROVIDER, CONNECTION_EXPIRY_TIME_MS } from '@/shared/config'
+
+const openStoreLink = () => {
+  const { currentNetwork } = useAppStore()
+  const browserName = browserInfo().id
+  const storeLink =
+    currentNetwork.storeUrls && currentNetwork.storeUrls[browserName]
+
+  window.open(storeLink, '_blank')
+}
+
+const setConnectionExpiry = () => {
+  const currentDate = Date.now()
+  const expiryDate = currentDate + CONNECTION_EXPIRY_TIME_MS
+
+  setItem(STORAGE_KEY.CONNECTION_EXPIRY, expiryDate.toString())
+}
+
+const connect = async () => {
+  const { showModal } = useModal()
+  const { formatMessage } = useIntl()
+  const { reloadProfile } = useProfileStore()
+  const { setIsConnected, reloadConnectedProfile } = useConnectionStore()
+
+  // when no extension installed we show modal
+  if (!INJECTED_PROVIDER) {
+    openStoreLink()
+
+    return showModal({
+      title: formatMessage('web3_connect_error_title'),
+      message: formatMessage('web3_connect_no_extension'),
+    })
+  }
+
+  try {
+    const { accounts, requestAccounts } = useWeb3(PROVIDERS.INJECTED)
+
+    let address = await accounts()
+
+    if (!address) {
+      ;[address] = await requestAccounts()
```

Comment:
oh idk this syntax `;[address]` what is it?

Reply:
Since `requestAccounts` return an array `;[address]` takes first element of array and assign to `address` variable